### PR TITLE
Fixed hardclip rendering, undeleted boxes when group is deleted

### PR DESCRIPTION
**Pull request by robinpaul85**
## Description
1) Fixed issue of rendering of hard clipped reads.
2) Pushed "Allele similarity" label upwards when alt allele is at the bottom of the page.
3) Deleted boxes in a group, when the group itself is being deleted.
4) Implemented filtering of supplementary alignment reads from UI (like how its implemented for PCR/optical duplicates).

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by akanksha-r — DISMISSED** [review]

**Review by karishma-gangwani — DISMISSED** [review]
looks good Robin.

File: client/src/block.tk.bam.js [hunk 1]
```diff
@@ -293,6 +294,7 @@ async function getData(tk, block, additional = {}) {
 	if (tk.indexURL) body.indexURL = tk.indexURL
 
 	if (tk.drop_pcrduplicates) body.drop_pcrduplicates = 1
+	if (tk.drop_supplementary_alignments) body.supplementary_alignments = 1
```
Comment (xzhou82) [inline client/src/block.tk.bam.js:297]: typo need to change property name in body

File: server/src/bam.js [hunk 2]
```diff
@@ -1855,18 +1867,31 @@ function parse_one_segment(arg, r, q) {
...
+				const b = {
 					opr: cigar,
 					start: pos,
 					len,
 					cidx: cum - len
-				})
+				}
+				if (segment.boxes.length == 0) {
+					// this is the first box, will not consume ref
+					// shift softclip start to left, so its end will be pos, will not increment pos
```
Comment (xzhou82) [inline server/src/bam.js:1878]: typo? is "hardclip" instead?

File: server/src/bam.js [hunk 3]
```diff
@@ -1855,18 +1867,31 @@ function parse_one_segment(arg, r, q) {
...
+					segment.boxes.push(b)
+				}
 				prev = i + 1
 				continue
 			}
 			if (cigar == 'N') {
 				// no seq
-			} else if (cigar == 'P' || cigar == 'D') {
+			} else if (cigar == 'P' || cigar == 'D' || cigar == 'H') {
 				// padding or del, no sequence in read
```
Comment (xzhou82) [inline server/src/bam.js:1895]: please modify comment that the ending hardclip is ignored

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**